app/src/main/java/edu/uwstout/p2pchat/NickNameFragment.java

```diff
@@ -0,0 +1,272 @@
+
+import java.util.ArrayList;
+
+
+
+/**
+ * NickNameFragment is the fragment responsible for showing the previously contacted
+ * nicknames.
+ * A simple {@link Fragment} subclass.
+ */
+public class NickNameFragment extends Fragment
+{
+    /**
+     * The listview that had been initilized in the xml file.
+     */
+    private ListView view;
+
+    /**
+     * context that the nickname fragment will be using.
+     */
+    private Context classContext;
+
+    /**
+     * Arraylist of names.
+     */
+    private ArrayList<String> names = new ArrayList<>();
+
+
+    private ArrayAdapter adapter;
+    /**
+     * Default constructor. Nothing will be used here.
+     */
+    public NickNameFragment()
+    {
+        //Nothing will be done here.
+    }
+
+
+    /**
+     *
+     * @param inflater inflator that came with with this method.
+     * @param container a container that came with this method.
+     * @param savedInstanceState just a bundle that came with this method.
+     * @return view for the nickname.
+     */
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container,
+            final Bundle savedInstanceState)
+    {
+        setNames();
+
+        //Inflate a view.
+        View view = inflater.inflate(R.layout.fragment_nick_name, container, false);
+
+        //Get the context from the view.
+        classContext = view.getContext();
+
+        //Put out on the screen that this is lauching.
+        Log.i("Launching", "Nickname view had launched");
+
+        //Inilizes listview for nicknames here.
+        ListView names = (ListView) view.findViewById(R.id.nameListView);
+
+        //Sets the adapter for the listview.
+        names.setAdapter(getNickNameAdapter());
+
+
+        setClickListener(names);
+
+        // Inflate the layout for this fragment
+        return view;
+    }
+
+    /** Creates an array adapter from the database and returns it.
+     * @return a newly created array adapter created from data.
+     */
+    private ArrayAdapter getNickNameAdapter()
+    {
+
+
+
+        //TODO get number of names from database here.
+        int size = 0; //Change later.
+
+        //TODO extract nicknames from database here.
+
+        //Adapter that will be used.
+        adapter = (new ArrayAdapter(classContext,
+                android.R.layout.simple_list_item_1,
+                names));
+
+        //Returns the new arrayadapter.
+        return adapter;
+    }
+
+
+    /**
+     * Seperate method to set the click listener for the.
+     */
+    private void setClickListener(ListView view)
+    {
+        //Onclick listener for all values of the list.
+        view.setOnItemClickListener(new AdapterView.OnItemClickListener()
+        {
+            @Override
+            public void onItemClick(final AdapterView<?> adapterView, View view, final int i,
+                    final long l)
+            {
+                Log.i("LOGGG", String.valueOf(i));
```

Comment:
this logging statement is both unhelpful if I were to dig through the log, and does not have an appropriate tag.